src/core/utils/useOnlineInternalHandleLoanReservation.ts

```diff
@@ -0,0 +1,112 @@
+import { useQueryClient } from "react-query";
+import {
+  getGetV1UserLoansQueryKey,
+  getGetV1UserReservationsQueryKey,
+  usePostV1UserLoansIdentifier,
+  usePostV1UserReservationsIdentifier
+} from "../../core/publizon/publizon";
+import { usePatronData } from "../../core/utils/helpers/usePatronData";
+import useReaderPlayer from "../../core/utils/useReaderPlayer";
+import { useUrls } from "../../core/utils/url";
+import { useModalButtonHandler } from "../../core/utils/modal";
+import { onlineInternalModalId } from "../../apps/material/helper";
+import { getAllFaustIds } from "../../core/utils/helpers/general";
+import { Manifestation } from "../../core/utils/types/entities";
+import { OnlineInternalRequestStatus } from "../../core/utils/types/request";
+import { CreateLoanResult } from "../publizon/model";
+
+type useOnlineInternalHandleLoanReservationType = {
+  manifestations: Manifestation[];
+  openModal: boolean;
+  setReservationStatus?: (status: OnlineInternalRequestStatus) => void;
+  setLoanResponse?: (response: CreateLoanResult | null) => void;
+};
+
+const useOnlineInternalHandleLoanReservation = ({
+  manifestations,
+  openModal,
+  setReservationStatus,
+  setLoanResponse
+}: useOnlineInternalHandleLoanReservationType) => {
```

Comment:
I find it really hard to determine the scope/purpose of a hook like this based on the current name and parameters alone. Perhaps it is just me.

Reply:
I'm open to any suggestions on alternative ways to handle this. Can we review it together?